web/src/email_pill.ts

```diff
@@ -0,0 +1,92 @@
+import * as blueslip from "./blueslip";
+import type {InputPillConfig, InputPillContainer, InputPillItem} from "./input_pill";
+import * as input_pill from "./input_pill";
+import type {User} from "./people";
+import * as people from "./people";
+
+type UserPill = {
+    email: string;
+};
+
+export type UserPillWidget = InputPillContainer<UserPill>;
+
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+export function create_item_from_email(
+    email: string,
+    current_items: InputPillItem<UserPill>[],
+): InputPillItem<UserPill> | undefined {
+    if (!emailRegex.test(email)) {
+        return undefined;
+    }
+
+    const existing_emails = current_items.map((item) => item.email);
+    if (existing_emails.includes(email)) {
+        return undefined;
+    }
+
+    return {
+        type: "user",
```

Comment:
Why isn't this a new `email` type? It seems like that would be more accurate than pretending that it's a user object.